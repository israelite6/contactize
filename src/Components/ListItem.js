import React, {useCallback} from 'react';
import PropTypes from 'prop-types';
import {Text, StyleSheet, TouchableOpacity} from 'react-native';

/**
 *
 * @param {string} label - Left Text
 * @param {string} value - Right Text
 */
export default function ListItem({label, value, onPress}) {
  return (
    <TouchableOpacity onPress={onPress} style={styles.root}>
      <Text>{label}</Text>
      <Text>{value}</Text>
    </TouchableOpacity>
  );
}

ListItem.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

const styles = StyleSheet.create({
  root: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 10,
  },
  image: {
    width: '100%',
    height: '100%',
  },
});
