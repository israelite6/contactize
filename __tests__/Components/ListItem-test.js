/**
 * @format
 */

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../../src/Components/ListItem';

it('ListItem component renders correctly', () => {
  const tree = renderer
    .create(<ListItem label="Mobile" value="Number" onPress={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
