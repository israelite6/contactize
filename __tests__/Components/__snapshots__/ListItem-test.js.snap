// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Text>
    Mobile
  </Text>
  <Text>
    Number
  </Text>
</View>
`;
